### PR TITLE
WIP: Include context path into skip pattern

### LINKED ISSUE #119: Skip pattern for multiple applications
In the spec we mentioned 

_Server side tracing can be disabled by specifying a skip pattern which is used to match with HTTP path UriInfo.getPath(). If the regex matches with HTTP path then tracing for the given server request is disabled even if the method or class is annotated with @Traced. The configuration does not disable any outbound request made from the disabled server endpoint._ 

UriInfo.getPath() does not include the context root of the app. What if we have 2 apps that have the same paths but just want to skip one but not the other.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #119 

* skip pattern has to include full URL without schema, host, port and query params e.g. `/context/app/endoint`
* test suite expects the test app to be deployed on `/opentracing` context root. 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Comment by pavolloffay** [comment]
@fmhwong could you please review? 

cc @eclipse/eclipse-microprofile

**Comment by pavolloffay** [comment]
Yet another solution is to require adding context root only if the app supports multiple deployments. Similar to this https://github.com/eclipse/microprofile-metrics/pull/108/files

File: spec/src/main/asciidoc/microprofile-opentracing.asciidoc [hunk 1]
```diff
@@ -184,13 +184,13 @@ If there is an exception object available the implementation SHOULD also add log
...
-The configuration does not disable any outbound request made from the disabled
-server endpoint.
+including context path and excluding query parameters e.g. `UriInfo.getAbsolutePath().getPath()`.
+If the regex matches with HTTP path then tracing for the given server request is disabled even if
+the method or class is annotated with `@Traced`. The configuration does not disable any outbound
+request made from the disabled server endpoint.
 
 The skip pattern is specified as a string with key `mp.opentracing.server.skip-pattern` which has to be
-compliant with `java.util.regex.Pattern`. An example skip pattern might be `mp.opentracing.server.skip-pattern=/health|/metrics.*`
+compliant with `java.util.regex.Pattern`. An example skip pattern might be `mp.opentracing.server.skip-pattern/context/=/health|/metrics.*`
```
Comment (fmhwong) [inline spec/src/main/asciidoc/microprofile-opentracing.asciidoc]: Is this suppose to be `mp.opentracing.server.skip-pattern=/context/health|/metrics.*`.  Actually, the MP endpoints are using "/" context root.  I suggest to change the example to `mp.opentracing.server.skip-pattern=/context/foo|/context/bar.*`

Comment (pavolloffay) [reply spec/src/main/asciidoc/microprofile-opentracing.asciidoc]: good point, fixed

**Comment by pavolloffay** [comment]
On hold until https://groups.google.com/forum/#!topic/microprofile/5sT6tgzvBic is resolved.

**Pull request closed without merging**